src/ctap/mod.rs

```diff
@@ -622,10 +598,23 @@ where
         ))
     }
 
+    // Generates a different per-credential secret for each UV mode.
+    // The computation is deterministic, and private_key expected to be unique.
+    fn generate_cred_random(
+        &mut self,
+        private_key: &crypto::ecdsa::SecKey,
+        has_uv: bool,
+    ) -> Result<[u8; 32], Ctap2StatusCode> {
+        let mut private_key_bytes = [0u8; 32];
+        private_key.to_bytes(&mut private_key_bytes);
+        let key = self.persistent_store.cred_random_secret(has_uv)?;
+        Ok(hmac_256::<Sha256>(&key, &private_key_bytes[..]))
```

Comment:
- This should be fine given the assumptions (i.e. `private_key_bytes` was chosen uniformly at random).
- Some nit on formatting: the `[..]` part is not really needed here. The only difference I see is that it creates a `&[u8]` rather than a `&[u8; 32]`, but the conversion to `&[u8]` is implicit IIRC.

Reply:
The assumption is correct. Removed the `[..]`.